### PR TITLE
fix throughput

### PR DESCRIPTION
**Pull request by Le-Zheng**
## What changes were proposed in this pull request?

fix throughput time count (+ count stage time)
do batch jobs by multi-thread `cache.localModels`

## How was this patch tested?

manul test + Jenkins test

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizerV2.scala [hunk 1]
```diff
@@ -202,14 +206,13 @@ object DistriOptimizerV2 extends AbstractOptimizer {
...
-
+      }.reduce(_ + _)
+    
     parameterSync(lossSum.value, successModels, cacheOfMaster, models, context)
-
+    val endTime = System.nanoTime() - startTime
+    
     driverStatesUpdate(cacheOfMaster, recordsNum.value,
-      context, trainingTrace, metrics)
+      context, trainingTrace, endTime, metrics)
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizerV2.scala:215]: why change the time counting?

Comment (Le-Zheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizerV2.scala:215]: each iterationTakes should be reduce() + parameterSync()
previous `trainingTrace.traceIteration()` can only hold reduce() time.

Comment (qiuxin2012) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizerV2.scala:215]: Can we add a new metric for parameter sync?

Comment (Le-Zheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizerV2.scala]: Done. Put `reduce() + parameterSync()` in `trainingTrace.traceIteration()` to count full time iteration takes.

**Comment by Le-Zheng** [comment]
Jenkins test passed http://10.239.47.210:18080/view/PR/job/BigDL-PRVN-lite/1155/ + http://10.239.47.210:18080/job/BigDL-PRVN-keras-py36/1175/

**Pull request merged**